cli/flox/src/commands/environment.rs

```diff
@@ -570,25 +609,202 @@ impl Activate {
             },
         };
 
-        if !self.run_args.is_empty() {
+        if !run_args.is_empty() {
             command.arg("-i");
             command.arg("-c");
-            command.arg(self.run_args.join(" "));
+            command.arg(run_args.join(" "));
         }
 
         debug!("running activation command: {:?}", command);
 
-        if self.run_args.is_empty() {
+        if run_args.is_empty() {
             let message = formatdoc! {"
                 ✅  You are now using the environment {now_active}.
                 To stop using this environment, type 'exit'"};
             info!("{message}");
         }
-        let error = command.exec();
-
         // exec should never return
+        command.exec().into()
+    }
+
+    /// Patch the PATH to undo the effects of `/usr/libexec/path_helper`
+    ///
+    /// Darwin has a "path_helper" which indiscriminately reorders the path
+    /// to put the system curated path items first in the `PATH`,
+    /// which completely breaks the user's ability to manage their `PATH` in subshells,
+    /// e.g. when using tmux.
+    ///
+    /// On macos `/usr/libexec/path_helper` is typically invoked from
+    /// default shell rc files, e.g. `/etc/profile` and `/etc/zprofile`.
+    ///
+    /// Note: since the "path_helper" i only invoked by login shells,
+    /// we only need to setup the PATH patching for `flox activate` in shell rc files.
```

Comment:
Fact check: is this statement correct? If it is we should prob revert https://github.com/flox/flox/pull/868/files#diff-e990d97da4f0228255a48f24ac11257011918ac0d5d57b2307efffd8ba382ea6R29-R44

Reply:
What if I:
```
$ flox activate
$ tmux # re-orders path
tmux$ flox activate
```
Wouldn't we want to re-order in that case?

Reply:
> but you never reverted the linked chunk of code

it was a fact check, basically me asking 🙃

Reply:
> ```
> $ flox activate
> $ tmux # re-orders path
> tmux$ flox activate
> ```
Do we need to change the comment because of this case?

Reply:
can we discuss this in the context of #889 to get this PR unblocked?

Reply:
Sure!